cmd/scollector/collectors/aws.go

```diff
@@ -0,0 +1,314 @@
+	awsStatusCheckFailed  string = "aws.ec2.status.failed"
+	awsELBLatencyMin      string = "aws.elb.latency.minimum"
+	awsELBLatencyMax      string = "aws.elb.latency.maximum"
+	awsELBLatencyAvg      string = "aws.elb.latency.average"
+	awsELBHostsHealthy    string = "aws.elb.hosts.healthy"
+	awsELBHostsUnHealthy  string = "aws.elb.hosts.unhealthy"
+	awsEC2CPUDesc         string = "The average CPU Utilization, gathered at a 60 second interval and averaged over five minutes."
+	awsEC2NetBytesDesc    string = "The average bytes transmitted or received via network, gathered at a 60 second interval and averaged over five minutes."
+	awsEC2DiskBytesDesc   string = "The average bytes written or read via disk, gathered at a 60 second interval and averaged over five minutes."
+	awsEC2DiskOpsDesc     string = "The average disk operations, either written or read, gathered at a 60 second interval and averaged over five minutes."
+	awsEC2StatusCheckDesc string = "The EC2 Status Check, which includes both instance-level and system-level drill-down, gathered every 60 seconds."
+	awsELBLatencyDesc     string = "The minimum, maximum and average latency as reported by the load balancer, gathered at a 60 second interval and averaged over five minutes."
+	awsELBHostCountDesc   string = "The number of instances in what the Elastic Load Balancer considers a healthy state, gathered every 60 seconds."
+)
+
+func AWS(accessKey, secretKey, region string) {
+	collectors = append(collectors, &IntervalCollector{
+		F: func() (opentsdb.MultiDataPoint, error) {
+			return c_aws(accessKey, secretKey, region)
+		},
+		Interval: 60 * time.Second,
+		name:     fmt.Sprintf("aws-%s", region),
+	})
+}
+
+func c_aws(accessKey, secretKey, region string) (opentsdb.MultiDataPoint, error) {
+	var md opentsdb.MultiDataPoint
+	creds := aws.Creds(accessKey, secretKey, "")
+	if creds == nil {
+		return nil, fmt.Errorf("Unable to make creds")
+	}
+	ecc := ec2.New(creds, region, nil)
+	if ecc == nil {
+		return nil, fmt.Errorf("Unable to login to EC2")
+	}
+	elb := elb.New(creds, region, nil)
+	if elb == nil {
+		return nil, fmt.Errorf("Unable to login to ELB")
+	}
+	cw := cloudwatch.New(creds, region, nil)
+	if cw == nil {
+		return nil, fmt.Errorf("Unable to login to CloudWatch")
+	}
+	instances, err := AWSGetInstances(*ecc)
+	if err != nil {
+		slog.Info("No EC2 Instances found.")
+	}
+	loadbalancers, err := AWSGetLoadBalancers(*elb)
+	if err != nil {
+		slog.Info("No ELB Load Balancecrs found.")
+	}
+	for _, loadbalancer := range loadbalancers {
+		AWSGetELBLatency(*cw, &md, loadbalancer)
+		AWSGetELBHostCounts(*cw, &md, loadbalancer)
+	}
+	for _, instance := range instances {
+		AWSGetCPU(*cw, &md, instance)
+		AWSGetNetwork(*cw, &md, instance)
+		AWSGetDiskBytes(*cw, &md, instance)
+		AWSGetDiskOps(*cw, &md, instance)
+		AWSGetStatusChecks(*cw, &md, instance)
+	}
+	return md, nil
+}
+
+func AWSGetInstances(ecc ec2.EC2) ([]ec2.Instance, error) {
```

Comment:
Fixed in 85794f0bb624956985af1e8cd57ee43367b5ff39

On Mon, Mar 9, 2015 at 4:09 PM, Matt Jibson <notifications@github.com>
wrote:

> In cmd/scollector/collectors/aws.go
> <https://github.com/bosun-monitor/bosun/pull/767#discussion_r26070385>:
>
> > +	}
> > +	for _, loadbalancer := range loadbalancers {
> > +		AWSGetELBLatency(*cw, &md, loadbalancer)
> > +		AWSGetELBHostCounts(*cw, &md, loadbalancer)
> > +	}
> > +	for _, instance := range instances {
> > +		AWSGetCPU(*cw, &md, instance)
> > +		AWSGetNetwork(*cw, &md, instance)
> > +		AWSGetDiskBytes(*cw, &md, instance)
> > +		AWSGetDiskOps(*cw, &md, instance)
> > +		AWSGetStatusChecks(*cw, &md, instance)
> > +	}
> > +	return md, nil
> > +}
> > +
> > +func AWSGetInstances(ecc ec2.EC2) ([]ec2.Instance, error) {
>
> Yes, these should all be awsXXX instead of AWSXXX so that they are
> unexported.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bosun-monitor/bosun/pull/767/files#r26070385>.
>